src/main/java/com/sbs/services/CouponService.java

```diff
@@ -0,0 +1,112 @@
+package com.sbs.services;
+
+import com.sbs.models.*;
+import com.sbs.repositories.CouponRepository;
+import com.sbs.repositories.MatchRepository;
+import com.sbs.utils.Exceptions.CouponInPlayException;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.sbs.models.CouponStatus.*;
+import static com.sbs.models.MatchResult.NOT_STARTED;
+
+@Service
+public class CouponService {
+    private final CouponRepository couponRepository;
+
+    private final MatchRepository matchRepository;
+
+    public CouponService(CouponRepository couponRepository, MatchRepository matchRepository) {
+        this.couponRepository = couponRepository;
+        this.matchRepository = matchRepository;
+    }
+
+    public Coupon addCoupon(Coupon coupon) {
+        coupon.setTotalCourse(1F);
+        coupon.setCouponStatus(CouponStatus.IN_PROGRESS);
+        return couponRepository.addCoupon(coupon);
+    }
+
+    public List<Coupon> getCoupons() {
+        return couponRepository.getCoupons();
+    }
+
+
+    public Optional<Coupon> addBet(Long couponId, Long matchId, String betName) throws CouponInPlayException {
+        Coupon coupon = couponRepository.findById(couponId)
+                .orElseThrow(() -> new HttpClientErrorException(HttpStatus.NOT_FOUND));
+        Match match = matchRepository.findById(matchId)
+                .orElseThrow(() -> new HttpClientErrorException(HttpStatus.NOT_FOUND));
+        if (IN_PROGRESS == coupon.getCouponStatus() && NOT_STARTED == match.getResult()) {
+            Bet bet = match.getOdds()
+                    .stream()
+                    .filter(b -> b.getName().equals(betName))
+                    .findFirst()
+                    .orElseThrow((() -> new HttpClientErrorException(HttpStatus.NOT_FOUND)));
```

Comment:
w wartstwie serwisow nie powinno sie sugerowac skad przyszedl obiekt. NOT_FOUND sugeruje ze to jest restowy kontroller. (a moze byc kolejkanp.). Stworz swoj customowy exception